src/CoherenceBuild/CoherenceBuild.cs

```diff
@@ -0,0 +1,165 @@
+            string apiKey)
+        {
+            _reposToProcess = reposToProcess;
+            _dropFolder = dropFolder;
+            _buildBranch = buildBranch;
+            _outputPath = outputPath;
+            _nugetPublishFeed = nugetPublishFeed;
+            _apiKey = apiKey;
+        }
+
+        public bool VerifyPackages { get; set; } = true;
+
+        public bool VerifyPartnerPackages { get; set; } = true;
+
+        public int Execute()
+        {
+            if (!Directory.Exists(_dropFolder))
+            {
+                Log.WriteError($"Drop share {_dropFolder} does not exist");
+                return 1;
+            }
+
+            Directory.CreateDirectory(_outputPath);
+            var processedPackages = new List<PackageInfo>();
+            foreach (var repo in _reposToProcess)
+            {
+                Program.Retry(() =>
+                {
+                    ProcessRepo(processedPackages, repo);
+                });
+            }
+
+            var coherenceVerify = new CoherenceVerifier(processedPackages, VerifyPackages, VerifyPartnerPackages);
+            if (!coherenceVerify.VerifyAll())
+            {
+                return 1;
+            }
+
+            if (!string.IsNullOrEmpty(_nugetPublishFeed))
+            {
+                PackagePublisher.PublishToFeed(processedPackages, _nugetPublishFeed, _apiKey);
+            }
+
+            return 0;
+        }
+
+        private void ProcessRepo(List<PackageInfo> processedPackages, RepoInfo repo)
+        {
+            var repoDirectory = Path.Combine(_dropFolder, repo.Name, _buildBranch);
+            if (string.IsNullOrEmpty(repo.BuildNumber))
+            {
+                repo.BuildNumber = FindLatest(repoDirectory);
+            }
+
+            repoDirectory = Path.Combine(repoDirectory, repo.BuildNumber);
+            foreach (var dependency in repo.FileSystemDependencies)
+            {
+                dependency.Copy(repoDirectory, _outputPath);
+            }
+
+            var packageSourceDir = Path.Combine(repoDirectory, repo.PackageSourceDir);
+
+            var packageTargetDir = Path.Combine(_outputPath, repo.PackageDestinationDir);
+            var symbolsTargetDir = Path.Combine(_outputPath, "symbols");
+            Directory.CreateDirectory(symbolsTargetDir);
+            Directory.CreateDirectory(packageTargetDir);
+
+            Parallel.ForEach(Directory.GetFiles(packageSourceDir, "*.nupkg"), packagePath =>
+            {
+                var packageFileName = Path.GetFileName(packagePath);
+                if (packagePath.EndsWith(".symbols.nupkg"))
+                {
+                    var targetPath = Path.Combine(symbolsTargetDir, packageFileName);
+                    File.Copy(packagePath, targetPath, overwrite: true);
+                    return;
+                }
+
+                var packageInfo = new PackageInfo
+                {
+                    IsPartnerPackage = repo.Name == "CoreCLR"
```

Comment:
Do we have ALL the partner packages coming from CoreCLR repo now?

(Also, can we rename this repo please? :smile:)